--- old/src/share/vm/memory/filemap.cpp	2016-10-20 11:02:27.892092470 -0200
+++ new/src/share/vm/memory/filemap.cpp	2016-10-20 11:02:27.824092538 -0200
@@ -263,7 +263,7 @@
         } else {
           struct stat st;
           if (os::stat(name, &st) == 0) {
-            if ((st.st_mode & S_IFDIR) == S_IFDIR) {
+            if ((st.st_mode & S_IFMT) == S_IFDIR) {
               if (!os::dir_is_empty(name)) {
                 ClassLoader::exit_with_path_failure(
                   "Cannot have non-empty directory in archived classpaths", name);
