--- old/src/share/vm/runtime/globals.hpp	2016-10-20 11:02:29.412090950 -0200
+++ new/src/share/vm/runtime/globals.hpp	2016-10-20 11:02:29.336091026 -0200
@@ -2887,9 +2887,6 @@
           "exceptions (0 means all)")                                       \
           range(0, max_jint/2)                                              \
                                                                             \
-  develop(bool, TraceStackWalk, false,                                      \
-          "Trace stack walking")                                            \
-                                                                            \
   /* notice: the max range value here is max_jint, not max_intx  */         \
   /* because of overflow issue                                   */         \
   diagnostic(intx, GuaranteedSafepointInterval, 1000,                       \
